shanoir-ng-datasets/src/main/java/org/shanoir/ng/studycard/model/condition/StudyCardDICOMConditionOnDatasets.java

```diff
@@ -0,0 +1,317 @@
+@JsonTypeName("StudyCardDICOMConditionOnDatasets")
+public class StudyCardDICOMConditionOnDatasets extends StudyCardCondition {
+	
+    private static final Logger LOG = LoggerFactory.getLogger(StudyCardDICOMConditionOnDatasets.class);
+    
+	private int dicomTag;
+
+    @Autowired
+    //WADODownloaderService downloader;
+	
+	public Integer getDicomTag() {
+        return dicomTag;
+    }
+
+    public void setDicomTag(Integer dicomTag) {
+        this.dicomTag = dicomTag;
+    }
+    
+    public boolean fulfilled(ExaminationAttributes<?> dicomAttributes) {
+        return fulfilled(dicomAttributes, new StringBuffer());
+    }
+
+    /**
+     * Check the conditions on a complete set of already known dicom Attributes
+     * @param <T> the type of the used keys
+     * @param examinationAttributes complete set of already known dicom Attributes, not a cache
+     * @param errorMsg
+     * @return
+     */
+    public <T> boolean fulfilled(ExaminationAttributes<T> examinationAttributes, StringBuffer errorMsg) {
+        if (examinationAttributes == null) throw new IllegalArgumentException("dicomAttributes can not be null");
+        int nbOk = 0; int total = 0; int nbUnknown = 0;
+        for (T acqId : examinationAttributes.getAcquisitionIds()) {
+            AcquisitionAttributes<T> acqAttributes = examinationAttributes.getAcquisitionAttributes(acqId);
+            for (T datasetId : acqAttributes.getDatasetIds()) {
+                total++;
+                boolean alreadyFulfilled = getCardinality() >= 1 && nbOk >= getCardinality();
+                if (!alreadyFulfilled) {
+                    Boolean fulfilled = fulfilled(acqAttributes.getDatasetAttributes(datasetId), errorMsg, datasetId );
+                    if (fulfilled == null) {
+                        nbUnknown++;
+                    }
+                    else if (fulfilled) {
+                        nbOk++;
+                    }
+                }  
+            }
+        }
+        boolean complies = cardinalityComplies(nbOk, nbUnknown, total);
+        writeConditionsReport(errorMsg, complies, nbOk, nbUnknown, total);
+        return complies;
+    }
+    
+    /**
+     * Check condition on acquisitions
+     * @param acquisitions data checked
+     * @param examinationAttributesCache to be used as a cache
+     * @param errorMsg
+     * @return
+     */
+    public boolean fulfilled(List<DatasetAcquisition> acquisitions, ExaminationAttributes<Long> examinationAttributesCache, WADODownloaderService downloader, StringBuffer errorMsg) {
+        if (acquisitions == null) throw new IllegalArgumentException("acquisitions can not be null");
+        int nbOk = 0; int total = 0; int nbUnknown = 0;
+        for (DatasetAcquisition acquisition : acquisitions) {
+            if (!examinationAttributesCache.has(acquisition.getId())) {
+                examinationAttributesCache.addAcquisitionAttributes(acquisition.getId(), new AcquisitionAttributes<Long>());
```

Comment:
Hi @julien-louis, I hope you are doing fine. I am not sure, your cache will work here, as you always only add a new and empty object to it?

Reply:
for the DatasetAttributes, it is different, it looks logic

Reply:
you are right, but as you have multiple acquisitions, you can not only add once:addAcquisitionAttributes(acquisition.getId(), new AcquisitionAttributes<Long>()); so you always add a new acq attributes to the cache?